webtau-docs/webtau/REST/matchers.md

```diff
@@ -5,86 +5,122 @@ type: two-sides
 # Response Mapping
 
 Identifiers inside validation closure are automatically mapped to a response body.  
+:include-empty-block: {rightSide: true}
 
-:include-groovy: com/twosigma/webtau/http/HttpTest.groovy {entry: "simple object mapping example", bodyOnly: true}
+:include-json: simpleObjectTestResponse.json {title: "object response", rightSide: false}
+
+```tabs {rightSide: true}
+Groovy: :include-groovy: com/twosigma/webtau/http/HttpGroovyTest.groovy {entry: "simple object mapping example", bodyOnly: true}
+Java: :include-java: com/twosigma/webtau/http/HttpJavaTest.java {entry: "simpleObjectMappingExample", bodyOnly: true}
+```
+
+:include-meta: {rightSide: false}
+
+List responses are handled by using index chain
 
 :include-meta: {rightSide: true}
-:include-json: simpleObjectTestResponse.json {title: "object response"}
+
+:include-empty-block: {rightSide: true}
 
 :include-meta: {rightSide: false}
 
-In case of an array response you need to access values using `body`.
 
-:include-groovy: com/twosigma/webtau/http/HttpTest.groovy {entry: "simple list mapping example", bodyOnly: true}
+:include-json: simpleListTestResponse.json {title: "list response"}
 
 :include-meta: {rightSide: true}
-:include-json: simpleListTestResponse.json {title: "list response"}
+
+```tabs
+Groovy: 
+
+:include-groovy: com/twosigma/webtau/http/HttpGroovyTest.groovy {entry: "simple list mapping example", bodyOnly: true}
+
+ Note: Groovy API implicitly assumes `body`, but if you need to deal with array response you need to access values using `body` explicitly.
+
+Java: :include-java: com/twosigma/webtau/http/HttpJavaTest.java {entry: "simpleListMappingExample", bodyOnly: true}
+```
 
 # Should and Should Not
 
 Matchers in webtau are triggered with `should` and `shouldNot` keywords.
+Additionally `shouldBe` and `shouldNotBe` alias keywords are available to make certain matcher combinations easier to read.
 
+`````tabs {rightSide: true}
+Groovy: 
 ```groovy
 myValue.should contain(10)
-myValue.shouldNot equal("hello")
-```  
-
-Some matchers have alternative shortcuts.
- 
-```groovy
-myValue.should != 10
-```  
+myValue.shouldNot == "hello"
+myValue.should != "hello" // alternative shortcut
+myValue.shouldBe > 10
+```
 
-Additionally `shouldBe` and `shouldNotBe` alias keywords are available to make certain matcher combinations easier to read
- 
-```groovy
+Java:
+```java
+myValue.should(contain(10))
+myValue.shouldNot(equal("hello"))
```

Comment:
`myValue.shouldBe greaterThan(10)` does not look like it would compile in Java 😬

Reply:
Was lazy and didn't move it to test files initially. Moving now.